ext/bcmath/bcmath.c

```diff
@@ -777,4 +791,943 @@ PHP_FUNCTION(bcscale)
+	}
+	return SUCCESS;
+}
+
+static zend_always_inline zend_result bcmath_number_mod_internal(
+	bc_num n1, bc_num n2, bc_num *ret,
+	size_t n1_full_scale, size_t n2_full_scale, size_t *scale, bool auto_scale
+) {
+	if (auto_scale) {
+		*scale = MAX(n1_full_scale, n2_full_scale);
+	}
+	if (!bc_modulo(n1, n2, ret, *scale)) {
+		zend_throw_exception_ex(zend_ce_division_by_zero_error, 0, "Modulo by zero");
+		return FAILURE;
+	}
+	bc_rm_trailing_zeros(*ret);
+	return SUCCESS;
+}
+
+static zend_result bcmath_number_pow_internal(
+	bc_num n1, bc_num n2, bc_num *ret,
+	size_t n1_full_scale, size_t *scale, bool auto_scale, bool is_op
+) {
+	/* Check the exponent for scale digits and convert to a long. */
+	if (UNEXPECTED(n2->n_scale != 0)) {
+		if (is_op) {
+			zend_value_error("exponent cannot have a fractional part");
+		} else {
+			zend_argument_value_error(1, "exponent cannot have a fractional part");
+		}
+		return FAILURE;
+	}
+	long exponent = bc_num2long(n2);
+
+	bool scale_expand = false;
+	if (auto_scale) {
+		if (exponent > 0) {
+			*scale = n1_full_scale * exponent;
+			if (UNEXPECTED(*scale > INT_MAX || *scale < n1_full_scale)) {
+				zend_value_error("scale of the result is too large");
+				return FAILURE;
+			}
+		} else if (exponent < 0) {
+			*scale = n1_full_scale + BC_MATH_NUMBER_EXPAND_SCALE;
+			if (UNEXPECTED(CHECK_RET_SCALE_OVERFLOW(*scale, n1_full_scale))) {
+				zend_value_error("scale of the result is too large");
+				return FAILURE;
+			}
+			scale_expand = true;
+		} else {
+			*scale = 0;
+		}
+	}
+
+	/**
+	 * bc_num2long() returns 0 if exponent is too large.
+	 * Here, if n2->n_value is not 0 but exponent is 0, it is considered too large.
+	 */
+	if (UNEXPECTED(exponent == 0 && (n2->n_len > 1 || n2->n_value[0] != 0))) {
+		if (is_op) {
+			zend_value_error("exponent is too large");
+		} else {
+			zend_argument_value_error(1, "exponent is too large");
+		}
+		return FAILURE;
+	}
+	bc_raise(n1, exponent, ret, *scale);
+	bc_rm_trailing_zeros(*ret);
+	if (scale_expand) {
+		size_t diff = *scale - (*ret)->n_scale;
+		*scale -= diff > BC_MATH_NUMBER_EXPAND_SCALE ? BC_MATH_NUMBER_EXPAND_SCALE : diff;
+	}
+	return SUCCESS;
+}
+
+static zend_always_inline bcmath_number_obj_t *bcmath_number_new_obj(bc_num ret, size_t scale)
+{
+	bcmath_number_obj_t *intern = get_bcmath_number_from_obj(bcmath_number_create(bcmath_number_ce));
+	intern->num = ret;
+	intern->scale = scale;
+	return intern;
+}
+
+static zend_result bcmath_number_parse_num(zval *zv, zend_object **obj, zend_string **str, zend_long *lval)
+{
+	if (Z_TYPE_P(zv) == IS_OBJECT && instanceof_function(Z_OBJCE_P(zv), bcmath_number_ce)) {
+		*obj = Z_OBJ_P(zv);
+		return SUCCESS;
+	} else {
+		switch (Z_TYPE_P(zv)) {
+			case IS_LONG:
+				*lval = Z_LVAL_P(zv);
+				return SUCCESS;
+
+			case IS_STRING:
+				*str = Z_STR_P(zv);
+				return SUCCESS;
+
+			case IS_NULL:
+				*lval = 0;
+				return SUCCESS;
```

Comment:
@Girgias 

This is for the same reason that there are problems with operators on GMP objects and null.

When merged this, it was intended to match the behavior of regular operators.

Since this behavior should be consistent, it would be best to discuss it together in my PR thread on GMP.

Reply:
ACK.